hannah-hjudge/Homework-3/object-homework/app.js

```diff
@@ -0,0 +1,113 @@
+/////////////////////////////////////////////////////////////
+
+///////////////////////// Challenge /////////////////////////
+
+/////////////////////////////////////////////////////////////
+
+// Write your answers inside this file, at the places where it's indicated by the comments.
+
+
+// 1. Suppose that we wanted to create a browser-based calendar program for keeping track of things. In comments, list at least three reasonable abstractions that you might use to build this program; for each abstraction, list out at least two properties and behaviors that it would make sense for that abstraction to have. Below is an example - please write your answer in the same format.
+  // 'Car'
+  // Description: Our app manages car sales.
+  // Every 'car' has
+  // - a make
+  // - a model
+  // - a year
+  // Every 'car' can
+  // - drive
+  // - brake
+  // - park
+
+// Answer Starts Here
+
+  // 'Book'
+  // Description: Our app manages book sales.
+  // Every 'book' has
+  // - a title
+  // - an author
+  // - a price
+  // Every 'book' can
+  // - be bought
+  // - be sold
+  // - be traded
+
+  // 'Dog'
+  // Description: Our app helps people find dogs for sale.
+  // Every 'dog' has
+  // - a breed
+  // - an age
+  // - a demeanor
+  // Every 'dog' can
+  // - bark
+  // - run
+  // - eat
+
+  // 'Train'
+  // Description: Our app manages train schedules.
+  // Every 'train' has
+  // - a route
+  // - a make
+  // - a service date
+  // Every 'train' can
+  // - drive
+  // - break down
+  // - arrive on time
+
+// Answer Ends Here
+
+
+// 2. Create an Object literal that lines up with the following description. Store it in the variable 'pet_owner', below. Be sure to give it reasonable values for each of its properties.
+
+  // 'Owner'
+  // Description: We are making an app for a veterinary clinic - it manages pets' vaccinations.
+  // Every 'owner' has:
+  // - a name
+  // - an address
+
+let pet_owner = {
+  name : "Sam",
+  address : "1 Market St",
+  alertOwner: function() {
+    console.log("Alert Sam on Jan 1 that his pet needs vaccinations.");
```

Comment:
How could you rewrite this to reference the value of the `name` parameter rather than hard coding `Sam`?